### ISSUE TITLE
Symfony Routing - Double forward slash at end causes fatal error

### ISSUE THREAD

**Issue by silverbackdan** [symfony/symfony#31270]
**Symfony version(s) affected**: 4.2.7

**Description**  
When loading a route which has a double trailing slash, `Symfony\Component\Routing\Matcher\Dumper\PhpMatcherTrait` throws an error `Notice: Undefined offset: 2` from here: https://github.com/symfony/symfony/blob/4.2/src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherTrait.php#L135

The `$vars` count is greater than the `$matches` array:
```
array:3 [▼
  0 => "/routes//"
  1 => "/"
  "MARK" => "334"
]
```

**How to reproduce**  
I am calling a resource from API Platform called `routes` but I've set the IDs to be the route path itself. So a front-end app knows what route it is at, and API platform then gets all the back-end components out for that route to return to the front-end. I think simply having a route with a double slash will cause this.

**Possible Solution**  
Not sure yet. I'll have to dig a bit deeper.

**Comment by silverbackdan** [comment]
Sorry I missed out what the `$vars` is populated as.
```
array:2 [▼
  0 => "id"
  1 => "_format"
]
```

So the ID should be `/` which is the route and I have configured routing to allow slashes as that parameter. With API Platform there is another route parameter `_format` which is optional but has a default.

No solution still but thought the additional information might help.

The route:
`api_routes_get_item                    GET            ANY      ANY    /routes/{id}.{_format}`

**Comment by silverbackdan** [comment]
@nicolas-grekas I have a feeling this is to do with your recent commits to this file dealing with trailing slashes unfortunately. Perhaps I'm better off leaving this with you as if I suggest a fix, it may not achieve whatever it was that you were trying to fix when these changes were made. Sorry for being the one to deliver the bad news about the recent updates!

**Comment by silverbackdan** [comment]
Sorry, the commits on 18th and 19th were after the 4.2.7 release. The master branch is failing right now if I try to use that because `PhpMatcherTrait` no longer exists so I imagine things are being updated and you may not need this PR.

**Comment by nicolas-grekas** [comment]
Isn't this solved already in 4.2.8@dev?

**Comment by nicolas-grekas** [comment]
Closing as duplicate of #31225, please report back if you think it's not the case.
